src/conn/conn_log.c

```diff
@@ -74,26 +90,56 @@ __logmgr_config(WT_SESSION_IMPL *session, const char **cfg, bool *runp)
 	if (cval.val != 0)
 		FLD_SET(conn->log_flags, WT_CONN_LOG_ARCHIVE);
 
-	WT_RET(__wt_config_gets(session, cfg, "log.file_max", &cval));
-	conn->log_file_max = (wt_off_t)cval.val;
-	WT_STAT_FAST_CONN_SET(session, log_max_filesize, conn->log_file_max);
+	if (!reconfig) {
+		/*
+		 * Ignore if the user tries to change the file size.  The
+		 * amount of memory allocated to the log slots may be based
+		 * on the log file size at creation and we don't want to
+		 * re-allocate that memory while running.
+		 */
+		WT_RET(__wt_config_gets(session, cfg, "log.file_max", &cval));
+		conn->log_file_max = (wt_off_t)cval.val;
+		WT_STAT_FAST_CONN_SET(session,
+		    log_max_filesize, conn->log_file_max);
+	}
 
-	WT_RET(__wt_config_gets(session, cfg, "log.prealloc", &cval));
 	/*
-	 * If pre-allocation is configured, set the initial number to one.
+	 * If pre-allocation is configured, set the initial number to a few.
 	 * We'll adapt as load dictates.
 	 */
+	WT_RET(__wt_config_gets(session, cfg, "log.prealloc", &cval));
 	if (cval.val != 0)
-		conn->log_prealloc = 1;
+		conn->log_prealloc = 5;
```

Comment:
@sueloverso, what's driving this change?  Won't preallocating 500MB of log files by default surprise some users?